openshift/filter.go

```diff
@@ -24,68 +24,88 @@ var availableKinds = []string{
 }
 
 type ResourceFilter struct {
-	Kinds []string
-	Name  string
-	Label string
+	Kinds          []string
+	Name           string
+	Label          string
+	ExcludedKinds  []string
+	ExcludedNames  []string
+	ExcludedLabels []string
 }
 
 // NewResourceFilter returns a filter based on kinds and flags.
 // kindArg might be blank, or a list of kinds (e.g. 'pvc,dc') or
 // a kind/name combination (e.g. 'dc/foo').
 // selectorFlag might be blank or a key and a label, e.g. 'name=foo'.
-func NewResourceFilter(kindArg string, selectorFlag string) (*ResourceFilter, error) {
+func NewResourceFilter(kindArg string, selectorFlag string, excludeFlag string) (*ResourceFilter, error) {
 	filter := &ResourceFilter{
 		Kinds: []string{},
 		Name:  "",
 		Label: selectorFlag,
 	}
 
-	if len(kindArg) == 0 {
-		return filter, nil
-	}
+	if len(kindArg) > 0 {
```

Comment:
Currently the resource parameter to include resources is handled differently than the exclude flag. Could/should the handling of the resource parameter be changed so that they work the same way as exclude (perhaps in a new issue)?

Reply:
Not entirely sure what you mean, maybe you can open an issue with the mentioned problem once I merge this? Thanks!